CMakeLists.txt

```diff
@@ -1,22 +1,29 @@
-cmake_minimum_required (VERSION 2.8.12)
+cmake_minimum_required (VERSION 3.25)
 project (libpd C)
 
+include(CMakeDependentOption)
+include(CheckLibraryExists)
+include(CheckIncludeFile)
+
 option(PD_UTILS  "Compile utilities" ON)
 option(PD_EXTRA  "Compile extras" ON)
 option(PD_MULTI  "Compile with multiple instance support" OFF)
 option(PD_LOCALE "Set the LC_NUMERIC number format to the default C locale" ON)
-option(PD_BUILD_C_EXAMPLES "Builds C API example programs" OFF)
+option(PD_DEFINE_EXTERN "Define EXTERN to a custom string for special linking purposes" "")
+option(LIBPD_STATIC "Build static library" ON)
+option(LIBPD_SHARED "Build shared library" ON)
+cmake_dependent_option(PD_BUILD_C_EXAMPLES "Builds C API example programs" OFF "LIBPD_SHARED" OFF)
 
 if (MSVC)
-    set(CMAKE_THREAD_LIBS_INIT CACHE PATH "Path to pthreads library binary file (ex. C:/src/vcpkg/packages/pthreads_x64-windows/lib/pthreadVC3.lib)")
+    set(PTHREADS_LIB CACHE PATH "Path to pthreads library binary file (ex. C:/src/vcpkg/packages/pthreads_x64-windows/lib/pthreadVC3.lib)")
```

Comment:
I guess this should be `FILEPATH` instead of `PATH` because we need to specify a file, not a directory.

Reply:
pushed!